### DISCUSSION TITLE
WIFI Management

### DISCUSSION THREAD

**Discussion by dtlubben** [CommunityToolkit/Maui#1135] [New Feature Discussions]
The Connectivity Object in Maui leaves a little bit to be desired.  What I am finding very difficult to work through is WIFI interfacing.

What would be great to have is a CommunityToolKit solution that would work across platforms that would Provide the following;

1) the Presently selected WIFI SSID Name.
2) the DNS Gateway IP address of #1.
3) A Scanned list of WIFI Names.
4) the Ability to switch to any Active Point I wish from #3.

Android had these capabilities and the deprecated some.  They stated for security reasons that it would throttle back #3 above, however it looks like it is gone.

They all are very tedious to work through.  It would be great to have the Communities help in making this a Cross Platform Solution.

Thanks,

Darrell

**Comment by TheCodeTraveler** [comment]
Thanks Darrell! Any suggestions on the implementation or the API design?

**Comment by dtlubben** [comment]
Thanks Brandon for your quick response.

Internet connectivity is the normal requirement of most of the applications out there, so knowing how it is connected is not important.  However, if you are connecting to External WIFI based devices the how and what are very important.

At a minimum the API would need to know WHAT it is connected to and what its IP address is, So:

1) Get_SSID_Name would return a String with the name of the Present Access Point selected on the Platform.
2) Get_Gateway_AP_IP would return a IPAddress type, which would contain the selected Access Point IP address.

Some Initialize and Status interfaces would probably be needed as the some platform interfaces include a CallBack to implement.

Example in Android
    NetworkRequest request = new NetworkRequest.Builder().AddTransportType(transportType: TransportType.Wifi).Build();
    ConnectivityManager connectivityManager = Android.App.Application.Context.GetSystemService(Context.ConnectivityService) as ConnectivityManager;
    NetworkCallbackFlags flagIncludeLocationInfo = NetworkCallbackFlags.IncludeLocationInfo;
    NetworkCallback networkCallback;

3) Event Notification would also be very helpful, to avoid Polling.

#1 and #2 above are relying on the user to select the Access Point.  When developing an application that communicates to an external device it would be nice to programmatically change the selected Access Point to known named device.  Instead of having the user have to know or be told to go to the Devices WIFI Settings and select the desired WIFI name (Access Point).  So:

#4 Set_SSID_Name which would be a String, that would change the selected Access Point.

When selection becomes available (#4), setting the Access Point Password would be needed to support newly added devices.
#5 Access_Point_Password of type string

All of the platforms keep track of what Access Points they have been connected to in the past.  This list normally gets updated with the Available Access Points that are available and in range of your Device.

#6 Get_Access_Point_List<> So a List of all Devices Past and New would be needed, a simple List of strings would not carry all of the information (active or not available) so some more complex List of Access Point structures would be needed.

Some interface to initiate a Scan of available devices might be necessary, as the underlying platform functions may only get the list upon request (just as when you open your WIFI manager, it takes a moment for the list to appear).

#7 Start_Access_Point_Scan()

#8 Event Handling for Change of Access_Point availability would be very helpful as well.

With all of the security and battery life issues these days, some platforms have deprecated some features and or throttled them back.  This impacts applications written to locate users at some store front providing Ads.  However, this should not be a problem for users simply wanting to pick their external WIFI device for their Mobile application connection.

Thank You for considering this addition to your Community solution.  Investigating all of the different solutions required for my application for the different platforms, I have found so many developers are having issues.  This would be a great MAUI Community addition.

Thanks,
Darrell

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
